org.eclipse.dawnsci.hdf5.model.test/src/org/eclipse/dawnsci/hdf5/model/internal/HierarchicalDataFileModelTest.java

```diff
@@ -42,46 +48,51 @@ public IHierarchicalDataFile getReader() throws Exception {
 	 */
 	@Test
 	public void printDataContents() throws Exception {
-		try (IHierarchicalDataFile reader = get_i05_4859_Reader.getReader()) {
+		try (NexusFile reader = get_i05_4859_Reader.getReader()) {
 			printDataContents(reader);
 		}
 
 	}
 
-	private void printDataContents(IHierarchicalDataFile reader)
-			throws Exception {
-		Map<String, Object> attributeValues = reader.getAttributeValues();
-		for (Entry<String, Object> entry : attributeValues.entrySet()) {
-			String ln = entry.getKey() + "=" + entry.getValue() + "=="
-					+ HierarchicalDataUtils.extractScalar(entry.getValue());
+	private void printDataContents(NexusFile reader) throws Exception {
+		String root = reader.getRoot();
+		GroupNode rootGroup = reader.getGroup(root, true);
+		Iterator<? extends Attribute> attrIter = rootGroup.getAttributeIterator();
+		while (attrIter.hasNext()) {
+			Attribute attribute = attrIter.next();
+			String ln = attribute.getName() + "=" + attribute.getValue() + "=="
+					+ NexusUtils.extractScalarFromDataset(attribute.getValue());
 			System.out.println(ln);
-
 		}
-
-		String g = reader.getRoot();
-		printGroup(reader, g);
+		printGroup(reader, root);
 	}
 
-	private void printGroup(IHierarchicalDataFile reader, String g) throws Exception {
-		
-		List<String> members = reader.memberList(g);
+	private void printGroup(NexusFile reader, String g) throws Exception {
+		GroupNode rootGroup = reader.getGroup("/", true);
+		List<String> members = (List<String>) rootGroup.getNames();
 
 		int n = members.size();
 		for (int i = 0; i < n; i++) {
 			String childPath = members.get(i);
+			childPath = Node.SEPARATOR + childPath;
 			System.out.print(childPath);
-			if (reader.isDataset(childPath)) {
-				hdf.object.Dataset dataset = (hdf.object.Dataset) reader.getData(childPath);
-				Object value = dataset.read();
+			GroupNode group = reader.getGroup(childPath, false);
+			if (group.isDataNode()) {
+				DataNode dataset = reader.getData(childPath);
+				Dataset value = DatasetUtils.sliceAndConvertLazyDataset(dataset.getDataset());
 				System.out.print("=DIMS(");
-				System.out.print(Arrays.toString(dataset.getDims()));
+				System.out.print(Arrays.toString(value.getShape()));
 				System.out.print(")");
-				System.out.print(value);
+				System.out.print(value.toString(true));
 				System.out.print("=="
-						+ HierarchicalDataUtils.extractScalar(value));
+						+ NexusUtils.extractScalarFromDataset(value));
 				
-			} else if (reader.isGroup(childPath)) {
-				printGroup(reader, childPath);
+			} else if (group.isGroupNode()) {
+				Collection<String> names = group.getNames();
+				for (Iterator<String> iterator = names.iterator(); iterator.hasNext();) {
+					String name = iterator.next();
+					System.out.println("node: " + name);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)